### PR TITLE
Add CCS based reindex in doc page

### LINKED ISSUE #116999: Add reindex via css to document
### Description

I would like to add reindex via ccs since it was not mentioned in the [official reindex reference](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html).

Reindex via ccs is a feature added at [this time](https://github.com/elastic/elasticsearch/pull/96968).

### PR DESCRIPTION
**Pull request by atsushi-matsui**
close #116999

Add CCS-based reindex to docs.

### CODE REVIEW

File: docs/changelog/117154.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 117154
+summary: "Add CCS based reindex to docs"
+area: Search
+type: bug
```
Comment (kingherc) [inline docs/changelog/117154.yaml:4]: This is not required since we're just adding documentation, and the original [PR](https://github.com/elastic/elasticsearch/pull/96968) had a changelog. Please remove this file from the PR.

File: docs/reference/docs/reindex.asciidoc [hunk 2]
```diff
@@ -1119,6 +1122,32 @@ POST _reindex
 // TEST[s/otherhost:9200/\${host}/]
 // TEST[s/\.\.\.,/"username": "test_admin", "password": "x-pack-test-password",/]
 
+[[CCS-based-reindex-from-remote]]
+==== CCS based reindex from remote
+
+Reindex supports CCS based reindexing from a remote Elasticsearch cluster:
```
Comment (kingherc) [inline docs/reference/docs/reindex.asciidoc:1128]: nit Would appreciate expanding CCS to Cross Cluster Search at least once in the text (like in this first mention here), and a link to the CCS section (`modules-cross-cluster-search`).

File: docs/reference/docs/reindex.asciidoc [hunk 3]
```diff
@@ -1119,6 +1122,32 @@ POST _reindex
...
+}
+--------------------------------------------------
+// TEST[setup:my_index_big]
+
+The `index` parameter must be specified with the remote cluster name and the index name, separated by 
+a colon (e.g., cluster_one:my-index-000001).
+Additionally, you must configure the remote cluster in advance and retrieve the remote cluster name.
+For more details, refer to the reference for <<add-remote-clusters>>.
+
+NOTE: Needs v8.12.0 or later on the reindexing cluster.
```
Comment (kingherc) [inline docs/reference/docs/reindex.asciidoc:1149]: This is insinuated by the documentation update in only recent versions, and the changelog in 8.12. So please remove this line.

**Review by kingherc — COMMENTED** [review]
Nice going! Added labels to automatically backport this to previous versions once approved and merged.

I think this looks fine to me in general. But I'd like to assign this to @ankikuma who's been more involved with reindex to take a look and finally approve, if possible.

File: docs/changelog/117154.yaml (hunk 1 above)
Comment (atsushi-matsui) [reply docs/changelog/117154.yaml]: [omit changelog](https://github.com/elastic/elasticsearch/pull/117154/commits/69985f7d920aa1eb2ea3f0e8e48f17269364f655)

File: docs/reference/docs/reindex.asciidoc (hunk 3 above)
Comment (atsushi-matsui) [reply docs/reference/docs/reindex.asciidoc]: [omit note](https://github.com/elastic/elasticsearch/pull/117154/commits/e0586516ce1d8415759d7ffbb1a97c6021b880be)

File: docs/reference/docs/reindex.asciidoc (hunk 2 above)
Comment (atsushi-matsui) [reply docs/reference/docs/reindex.asciidoc]: [expand and use lowercase](https://github.com/elastic/elasticsearch/pull/117154/commits/34a6708404200dba0aad0371863fb34197779333)

**Comment by atsushi-matsui** [comment]
@kingherc 
Thank you for your comment. I have corrected it, so please check it.

**Review by kingherc — COMMENTED** [review]
Thanks! Looks good to me. @ankikuma would you be able to handle final review & approval?

**Comment by ankikuma** [comment]
Based on this [PR](https://github.com/elastic/elasticsearch/pull/96968), I think we might also need to mention ignore_unavailable and skip_unavailable settings, for a more comprehensive documentation. Adding Albert to the review since he reviewed the PR that added support for reindex over CCS.

File: docs/reference/docs/reindex.asciidoc [hunk 4]
```diff
@@ -1119,6 +1122,30 @@ POST _reindex
 // TEST[s/otherhost:9200/\${host}/]
 // TEST[s/\.\.\.,/"username": "test_admin", "password": "x-pack-test-password",/]
 
+[[ccs-based-reindex-from-remote]]
+==== ccs based reindex from remote
+
+Reindex supports Cross-cluster search (<<modules-cross-cluster-search,{ccs}>>) based reindexing from a remote Elasticsearch cluster:
```
Comment (albertzaharovits) [inline docs/reference/docs/reindex.asciidoc:1128]:
```suggestion

[[reindex-from-remote-cluster]]
==== reindex from remote cluster

Reindex supports sourcing documents from a remote Elasticsearch cluster:
```

File: docs/reference/docs/reindex.asciidoc [hunk 5]
```diff
@@ -1119,6 +1122,30 @@ POST _reindex
...
+  "dest": {
+    "index": "my-new-index-000001"
+  }
+}
+--------------------------------------------------
+// TEST[setup:my_index_big]
+
+The `index` parameter must be specified with the remote cluster name and the index name, separated by 
+a colon (e.g., cluster_one:my-index-000001).
+Additionally, you must configure the remote cluster in advance and retrieve the remote cluster name.
```
Comment (albertzaharovits) [inline docs/reference/docs/reindex.asciidoc:1146]: I don't follow what "retrieve the remote cluster name" means here.
Perhaps, it should all be rephrased more simply such as: "Note that reindexing requires that the remote cluster be configured on the local cluster."

**Review by albertzaharovits — COMMENTED** [review]
Thank you for your contribution @atsushi-matsui !
I've left a few nitpicks to avoid the "ccs-based" description when referring to reindex from remote (I think this is a confusing low-level detail).

I'm OK to merge after it has been addressed.

File: docs/reference/docs/reindex.asciidoc (hunk 5 above)
Comment (atsushi-matsui) [reply docs/reference/docs/reindex.asciidoc:1146]: That certainly seems confusing.
I'll rephrase it to a simpler version.

File: docs/reference/docs/reindex.asciidoc (hunk 4 above)
Comment (atsushi-matsui) [reply docs/reference/docs/reindex.asciidoc:1128]: Since a section named [reindex-from-remote](https://github.com/elastic/elasticsearch/blob/2895f1e900b2f41704fd507845102a281cff437e/docs/reference/docs/reindex.asciidoc?plain=1#L958) already exists, won't it be difficult to understand what the difference is with `reindex-from-remote-cluster`?
You said that ccs is a low-level feature and we don't need to worry about it too much, but I think it would be better to clarify the difference with the existing remote reindex.

File: docs/reference/docs/reindex.asciidoc [hunk 6]
```diff
@@ -92,6 +92,9 @@ privilege for the source data stream, index, or alias.
 in the `reindex.remote.whitelist` setting of `elasticsearch.yml`. See
 <<reindex-from-remote>>.
 
+* If ccs-based reindexing from a remote cluster, you must add the remote cluster setting. See
```
[the remark this answers is not in the document]
Comment (atsushi-matsui) [reply docs/reference/docs/reindex.asciidoc:95]: Since it looks like a condition for performing an [existing reindex](https://github.com/elastic/elasticsearch/blob/2895f1e900b2f41704fd507845102a281cff437e/docs/reference/docs/reindex.asciidoc?plain=1#L91), I think it would be better to give it a different name to avoid confusing users.

File: docs/reference/docs/reindex.asciidoc (hunk 5 above)
Comment (atsushi-matsui) [reply docs/reference/docs/reindex.asciidoc]: [simple note](https://github.com/elastic/elasticsearch/pull/117154/commits/6ae57e05cdb51ec530aab31bcdc1cf7a7a987eca)

**Comment by atsushi-matsui** [comment]
@albertzaharovits 
Thank you for suggesting the correction.
I made a comment because there was something that bothered me a little, but if there are no problems, I will correct it again based on the suggestion.
I thought that omitting the naming "ccs-based" would cause confusion due to the difference with the existing remote reindex.

File: docs/reference/docs/reindex.asciidoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (ankikuma) [reply docs/reference/docs/reindex.asciidoc:95]: Yes I agree that we should keep the terminology to "ccs-based reindexing" to distinguish this from the pre-existing functionality of reindexing from a remote cluster.

**Comment by ankikuma** [comment]
I think it's important to distinguish the CCS based approach from the HTTP based reindexing from remote, also based on this [comment](https://github.com/elastic/elasticsearch/pull/96968#pullrequestreview-1668740949) from David. This looks good to me.

**Review by ankikuma — APPROVED** [review]

**Comment by leemthompo** [comment]
run docs-build

**Comment by leemthompo** [comment]
@elasticsearchmachine test this

**Comment by leemthompo** [comment]
> [!IMPORTANT]  
> Elastic documentation is migrating to Markdown for version 9.0+. See the [migration guide](https://elastic.github.io/docs-builder/migration/freeze/index.html) for details.


## ℹ️ **What's happening?**
- Starting January 29, we will start closing all unmerged documentation PRs targeting main/master
- We're migrating from AsciiDoc to Markdown for 9.0+
- 9.0 docs will be frozen from January 29 until February 20 2024
- **NOTE**: PRs that include both code and documentation changes will remain open

### **What do I need to do?**

#### For **<=8.x docs**:
 1. Rebase your PR to target the relevant 8.x branch instead
 2. The content can remain in AsciiDoc format

#### For **9.0+ docs**:

_Option 1_:
 - Draft docs in Markdown
 - Once migration freeze ends, find the relevant page in the new docs system and use the edit options to submit your changes


_Option 2_:
- Open an issue in [elastic/docs-content](https://github.com/elastic/docs-content)

### 💡 **Need help?**
1. For Elasticians: Ask in **#docs** Slack channel
2. For external contributors: Open an issue in [elastic/docs-content](https://github.com/elastic/docs-content)

**Comment by leemthompo** [comment]
@ankikuma @atsushi-matsui @albertzaharovits Please merge this PR today before the migration freeze. I just merged main, but there was a docs-check CI failure on previous commit that might require investigating. Thanks!
